src/DecoratorManifold.jl

```diff
@@ -0,0 +1,475 @@
+        function ($fname)($(callargs...); $(kwargs_list...)) where {$(where_exprs...)}
+            return ($fname)($(argnames[1]), _acts_transparently($fname, $(argnames...)), $(argnames[2:end]...),; $(kwargs_list...))
+        end
+        function ($fname)($(callargs[1]), ::Val{:transparent}, $(callargs[2:end]...); $(kwargs_list...)) where {$(where_exprs...)}
+            return ($fname)($(argnames[1]).manifold, $(argnames[2:end]...); $(kwargs_list...))
+        end
+        function ($fname)($(callargs[1]), ::Val{:intransparent}, $(callargs[2:end]...); $(kwargs_list...)) where {$(where_exprs...)}
+            error(manifold_function_not_implemented_message($(argnames[1]), $fname, $(argnames[2:end]...)))
+        end
+        function ($fname)($(callargs[1]), ::Val{:parent}, $(callargs[2:end]...); $(kwargs_list...)) where {$(where_exprs...)}
+            return invoke($fname, Tuple{supertype($(argtypes[1])), $(argtypes[2:end]...)}, $(argnames...); $(kwargs_list...))
+        end
+    end)
+end
+
+#
+# Functions
+#
+
+"""
+    is_default_decorator(M)
+
+For any manifold that is a subtype of [`AbstractDecoratorManifold`](@ref), this function
+indicates whether a certain manifold `M` acts as a default decorator.
+
+This yields that _all_ functions are passed through to the decorated [`Manifold`](@ref)
+if `M` is indicated as default. This overwrites all [`is_decorator_transparent`](@ref)
+values.
+
+This yields the following advantange: For a manifold one usually implicitly assumes for
+example a metric. To avoid reimplementation of this metric when introducing a second metric,
+the first metric can be set to be the default, i.e. its implementaion is already given by
+the undecorated case.
+
+to change this value, see [`default_decorator_dispatch`](@ref).
+"""
+is_default_decorator(M::Manifold) = _extract_val(default_decorator_dispatch(M))
+
+"""
+    default_decorator_dispatch(M)
+
+A function to decide whether by default to dispatch th the inner manifold of
+a decorator (`Val(true)`) or not (`Val(false`). For more details see
+[`is_decorator_transparent`](@ref).
+"""
+default_decorator_dispatch(M::Manifold) = Val(false)
+
+"""
+    is_decorator_transparent(f, M, args...)
+
+Given a [`Manifold`](@ref) `M` and a function `f(M,arge...)`, indicate, whether a
+[`AbstractDecoratorManifold`](@ref) acts transparent for `f`. This means, it
+just passes through down to the internally stored manifold.
+Only decorator manifolds can be transparent and their default is, to be transparent.
+A function that is affected by the decorator indicates this by returning `false`. To change
+this behaviour, see [`decorator_transparent_dispatch`](@ref).
+
+If a decorator manifold is not in general transparent, it might still pass down
+for the case that a decorator is the default decorator, see [`is_default_decorator`](@ref).
+"""
+function is_decorator_transparent(f, M::Manifold, args...)
+    return decorator_transparent_dispatch(f, M, args...) == Val(:transparent)
```

Comment:
```suggestion
    return decorator_transparent_dispatch(f, M, args...) === Val(:transparent)
```
This should also work and could be faster.